### PR TITLE
fix(TDOPS-3569/webpack): fix sass hmr on file change

### PR DESCRIPTION
**Pull request by Gbacc**
**What is the problem this PR is trying to solve?**
Styles dont get updated in dev mode when a sass file is changed

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: tools/scripts-config-react-webpack/config/webpack.config.umd.js [hunk 1]
```diff
@@ -27,6 +27,8 @@ module.exports = options => {
 	const userSassData = options.getUserConfig('sass', {});
 	const sassData = getSassData(userSassData);
 	const isEnvProd = options.mode === 'production';
+	const isEnvDevelopmentServe =
+		options.mode === 'development' && process.env.WEBPACK_SERVE === 'true';
```
Comment (jmfrancois) [inline tools/scripts-config-react-webpack/config/webpack.config.umd.js:31]: where WEBPACK_SERVE come from ?

Comment (Gbacc) [reply tools/scripts-config-react-webpack/config/webpack.config.umd.js:31]: It's an env variable , described here : https://webpack.js.org/api/cli/#environment-options

Comment (Gbacc) [reply tools/scripts-config-react-webpack/config/webpack.config.umd.js:31]: image

Comment (jmfrancois) [reply tools/scripts-config-react-webpack/config/webpack.config.umd.js:31]: this is the UMD config, so it is never used with serve

Comment (Gbacc) [reply tools/scripts-config-react-webpack/config/webpack.config.umd.js:31]: Ho yes that's right , i can just pass false

File: tools/scripts-config-react-webpack/config/webpack.config.common.js [hunk 2]
```diff
@@ -38,8 +38,9 @@ function getCommonStyleLoaders(enableModules, mode) {
 			importLoaders: 1,
 		};
 	}
+	const styleLoader = isEnvDevelopmentServe ? 'style-loader' : MiniCssExtractPlugin.loader;
 	return [
-		{ loader: MiniCssExtractPlugin.loader, options: { esModule: false } },
+		{ loader: styleLoader, options: { esModule: false } },
```
Comment (jmfrancois) [inline tools/scripts-config-react-webpack/config/webpack.config.common.js:43]: :+1:

File: tools/scripts-config-react-webpack/config/webpack.config.umd.js (hunk 1 above)
Comment (jmfrancois) [inline tools/scripts-config-react-webpack/config/webpack.config.umd.js]:
```suggestion
	const isEnvDevelopmentServe = false;
```

**Review by jmfrancois — APPROVED** [review]

**Review by romainseb — APPROVED** [review]

**Pull request merged**